qualtran/bloqs/chemistry/pbc/first_quantization/select_t.py

```diff
@@ -0,0 +1,65 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+r"""Bloqs for SELECT T for the first quantized chemistry Hamiltonian."""
+from functools import cached_property
+from typing import Set, TYPE_CHECKING
+
+from attrs import frozen
+
+from qualtran import Bloq, Register, Signature
+from qualtran.bloqs.basic_gates import Toffoli
+
+if TYPE_CHECKING:
+    from qualtran.resource_counting import BloqCountT, SympySymbolAllocator
+
+
+@frozen
+class SelectTFirstQuantization(Bloq):
+    r"""SELECT for the kinetic energy operator for the first quantized chemistry Hamiltonian.
+
+    Args:
+        num_bits_p: The number of bits to represent each dimension of the momentum register.
+        eta: The number of electrons.
+
+    Registers:
+        sys: The system register.
+        plus: A $|+\rangle$ state.
+        flag_T: a flag to control on the success of the $T$ state preparation.
+
+    References:
+        [Fault-Tolerant Quantum Simulations of Chemistry in First Quantization](
+            https://arxiv.org/abs/2105.12767) page 20, section B
+    """
+    num_bits_p: int
+    eta: int
+
+    @cached_property
+    def signature(self) -> Signature:
+        return Signature(
+            [
+                Register("sys", bitsize=self.num_bits_p, shape=(self.eta, 3)),
```

Comment:
this can result in a lot of soquets, right? I think the ultimate strategy will be to do something like #300 where we have a high-level interface bloq that has two registers "select" and "system" and then implementation bloqs that has the parts all factored into meaningful components but there's a lot of soquets and you use a `Partition` bloq to go betwixt the two. Out of the scope for this PR but I don't have any actionable comments, so I figured I'd write this comment here.

Reply:
right, I've been sort of running into this pain in https://github.com/quantumlib/Qualtran/pull/493/files#diff-c6f700b3dd40e8b12b6388c73fc9700b8331972cd0a508392cda5687a07acef4R338

Reply:
It would be helpful to consolidate some of these things with the existing SelectOracle and PrepareOracle.